Level-Y/code.go

```diff
@@ -0,0 +1,91 @@
+///////////////////////////////////////////////////////////////////
+///                                                      		///
+///   0. code_test.go is passing but the code is vulnerable  	///
+///   1. Review the code. Can you spot the bugs(s)?       		///
+///   2. Fix the code.go but ensure that code_test.go passes    ///
+///   3. Run hack_test.go and if passing then CONGRATS!       	///
+///   4. If stuck then read the hint                     		///
+///   5. Compare your solution with solution/solution.go   		///
+///                                                      		///
+///////////////////////////////////////////////////////////////////
+
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"regexp"
+)
+
+// Test users
+var testFakeMockUsers = map[string]string{
+	"user1@example.com": "password12345",
+	"user2@example.com": "B7rx9OkWVdx13$QF6Imq",
+	"user3@example.com": "hoxnNT4g&ER0&9Nz0pLO",
+	"user4@example.com": "Log4Fun",
+}
+
+var reqBody struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
```

Comment:
I would keep the struct here but I would create a variable locally inside `loginHandler`. Trying to avoid global variable as much as possible.

Reply:
Thank you for the comment.
I made the changes: https://github.com/skills/secure-code-game/commit/fb9b147bf3616102b93988bcebb9fd86ec9f24ae#diff-5959c2f73afa0079b89f4a37671b2883fd327a4df13053d0415ade8d5bb0d96cR40